streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -180,70 +159,90 @@ Sensor sourceNodeForwardSensor() {
-                context.taskId().toString(),
-                "processor-node",
+
+            nodeCreationSensor = createTaskAndNodeLatencyAndThroughputSensors(
+                "create",
+                metrics,
+                group,
+                taskName,
                 processorNodeName,
-                "destroy",
-                Sensor.RecordingLevel.DEBUG,
-                "task-id", context.taskId().toString()
+                allTagMap,
+                tagMap
             );
-            this.sourceNodeForwardSensor = metrics.addThroughputSensor(
-                context.taskId().toString(),
-                "processor-node",
+
+            // note: this metric can be removed in the future, as it is only recorded before being immediately removed
+            nodeDestructionSensor = createTaskAndNodeLatencyAndThroughputSensors(
+                "destroy",
+                metrics,
+                group,
+                taskName,
                 processorNodeName,
+                allTagMap,
+                tagMap
+            );
+
+            sourceNodeForwardSensor = createTaskAndNodeLatencyAndThroughputSensors(
                 "forward",
-                Sensor.RecordingLevel.DEBUG,
-                "task-id", context.taskId().toString()
+                metrics,
+                group,
+                taskName,
+                processorNodeName,
+                allTagMap,
+                tagMap
             );
+
+            this.taskName = taskName;
+            this.processorNodeName = processorNodeName;
         }
 
         private void removeAllSensors() {
-            metrics.removeSensor(nodeProcessTimeSensor);
-            metrics.removeSensor(nodePunctuateTimeSensor);
-            metrics.removeSensor(sourceNodeForwardSensor);
-            metrics.removeSensor(nodeCreationSensor);
-            metrics.removeSensor(nodeDestructionSensor);
+            metrics.removeAllNodeLevelSensors(taskName, processorNodeName);
         }
-    }
 
-    private static void runAndMeasureLatency(final Time time, final Runnable action, final Sensor sensor) {
-        long startNs = -1;
-        if (sensor.shouldRecord()) {
-            startNs = time.nanoseconds();
-        }
-        action.run();
-        if (startNs != -1) {
-            sensor.record(time.nanoseconds() - startNs);
+        private static Sensor createTaskAndNodeLatencyAndThroughputSensors(final String operation,
+                                                                           final StreamsMetricsImpl metrics,
+                                                                           final String group,
+                                                                           final String taskName,
+                                                                           final String processorNodeName,
+                                                                           final Map<String, String> taskTags,
```

Comment:
The last two parameters name are confusing: should they just be `allTagMap` and `tagMap`?

Reply:
Note that although processor-level metrics parent with "task=[taskName], processor-node-id=all" is actually the same for task-level `task=[taskName]`, today we still record them as two sensors (I have a proposal for cleaning up them in the future).

Reply:
I chose those names to match the method name: `createTaskAndNodeLatencyAndThroughputSensors`. Since we are creating "task" and "node" sensors, we pass in "task" and "node tags.

I felt that `allTagMap` and `tagMap` weren't self-expanatory enough in this interface.

Note that previously, we constructed maps with those names in a more generic context, where we didn't know, for example, that the child sensor was actually a "node" sensor, so the more generic names made more sense in that context.

If you're proposing to remove the `processor-node-id=all` tag in the future, I would support that.

Reply:
Yes I do propose it, but it would require a KIP, and I'd want to keep this PR small :)